pkg/repository/base.go

```diff
@@ -147,6 +147,7 @@ type EventRepository interface {
 	PushContainerStoppedEvent(containerID string, workerID string)
 	PushWorkerStartedEvent(workerID string)
 	PushWorkerStoppedEvent(workerID string)
+	PushAbstractionTriggeredEvent(workspaceId string, stub *types.Stub)
```

Comment:
I don't really like this name, it feels confusing. I'd almost argue for more specific functions even at the risk of more code / being more verbose. I./e. `PushDeployedStubEvent()` `PushServedStubEvent()`

Reply:
I like this better. I renamed the subject from abstraction -> stub